flink-staging/flink-ml/src/main/scala/org/apache/flink/ml/preprocessing/StandardScaler.scala

```diff
@@ -0,0 +1,178 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.preprocessing
+
+import java.lang.Iterable
+import breeze.linalg
+import breeze.linalg.DenseVector
+import breeze.numerics.sqrt
+import breeze.numerics.sqrt._
+import org.apache.flink.api.common.functions._
+import org.apache.flink.api.scala._
+import org.apache.flink.configuration.Configuration
+import org.apache.flink.ml.common.{Parameter, ParameterMap, Transformer}
+import org.apache.flink.ml.math.Breeze._
+import org.apache.flink.ml.math.Vector
+import org.apache.flink.ml.preprocessing.StandardScaler.{Mean, Std}
+import org.apache.flink.util.Collector
+
+/** Scales observations, so that all features have mean equal to zero
+  * and standard deviation equal to one
+  *
+  * This transformer takes a a Vector of values and maps it into the
+  * scaled Vector that each feature has a user-specified mean and standard deviation.
+  *
+  * This transformer can be prepended to all [[Transformer]] and
+  * [[org.apache.flink.ml.common.Learner]] implementations which expect an input of
+  * [[Vector]].
+  *
+  * @example
+  * {{{
+  *                              val trainingDS: DataSet[Vector] = env.fromCollection(data)
+  *
+  *                              val transformer = StandardScaler().setMean(10.0).setStd(2.0)
+  *
+  *                              transformer.transform(trainingDS)
+  * }}}
+  *
+  * =Parameters=
+  *
+  * - [[StandardScaler.Mean]]: The mean value of transformed data set; by default equal to 0
+  * - [[StandardScaler.Std]]: The standard deviation of the transformed data set; by default
+  * equal to 1
+  */
+class StandardScaler extends Transformer[Vector, Vector] with Serializable {
+
+  def setMean(mu: Double): StandardScaler = {
+    parameters.add(Mean, mu)
+    this
+  }
+
+  def setStd(std: Double): StandardScaler = {
+    parameters.add(Std, std)
+    this
+  }
+
+  override def transform(input: DataSet[Vector], parameters: ParameterMap):
+  DataSet[Vector] = {
+    val resultingParameters = this.parameters ++ parameters
+    val mean = resultingParameters(Mean)
+    val std = resultingParameters(Std)
+
+    val featureMetrics = extractFeatureMetrics(input)
+
+    input.map(new RichMapFunction[Vector, Vector]() {
+
+      var broadcastMeanSet: Vector = null
+      var broadcastStdSet: Vector = null
+
+      override def open(parameters: Configuration): Unit = {
+        val broadcastedMetrics = getRuntimeContext().getBroadcastVariable[(Vector,
+          Vector)]("broadcastedMetrics").get(0)
+        broadcastMeanSet = broadcastedMetrics._1
+        broadcastStdSet = broadcastedMetrics._2
+      }
+
+      override def map(vector: Vector): Vector = {
+        var myVector = vector.asBreeze
+
+        myVector :-= broadcastMeanSet.asBreeze
+        myVector :/= broadcastStdSet.asBreeze
```

Comment:
By storing the ```broadcastMeanSet``` and ```broadcastStdSet``` directly as ```BreezeVector``` we can avoid the repetitive wrapping of ```Vector``` into a ```BreezeVector``` instance.